src/libasr/asdl_cpp.py

```diff
@@ -699,38 +700,38 @@ def visitConstructor(self, cons, _):
         self.make_visitor(cons.name, cons.fields, True)
 
     def make_visitor(self, name, fields, cons):
-        self.emit("void visit_%s(const %s_t &x) {" % (name, name), 1)
-        self.emit(          'if(!attached) {', 2)
-        self.emit(              'if(start_line) {', 3)
-        self.emit(                  'start_line = false;', 4)
-        self.emit(                  's.append(indtd);', 4)
-        self.emit(              '} else {', 3)
-        self.emit(                  's.append("\\n"+indtd);', 4)
-        self.emit(              '}', 3)
-        self.emit(              'last ? s.append("└-") : s.append("|-");', 3)
-        self.emit(          '}', 2)
-        self.emit(          'last ? inc_indent() : inc_lindent();', 2)
-        self.emit(          'attached = true;', 2)
-        self.emit(          'last = false;', 2)
-        if cons:
-            self.emit(    'if (use_colors) {', 2)
-            self.emit(        's.append(color(style::bold));', 3)
-            self.emit(        's.append(color(fg::magenta));', 3)
-            self.emit(    '}', 2)
-            self.emit(    's.append("%s");' % name, 2)
-            self.emit(    'if (use_colors) {', 2)
-            self.emit(        's.append(color(fg::reset));', 3)
-            self.emit(        's.append(color(style::reset));', 3)
-            self.emit(    '}', 2)
-        self.used = False
-        for n, field in enumerate(fields):
-            self.visitField(field, cons, n == len(fields)-1)
-        self.emit(    'dec_indent();', 2)
-        if not self.used:
-            # Note: a better solution would be to change `&x` to `& /* x */`
-            # above, but we would need to change emit to return a string.
-            self.emit("if ((bool&)x) { } // Suppress unused warning", 2)
-        self.emit("}", 1)
+        if len(fields) == 2 and (name == "TranslationUnit" or name == "Module"):
+            self.emit("void visit_%s(const %s_t &/*x*/) {" % (name, name), 1)
+            self.emit("// Handled in the pickle.cpp", 2)
+            self.emit("}", 1)
```

Comment:
Here, the `TranslationUnit` is for ASR, and the `Module` is for LPython AST.
This fixes the issue that the nodes are visited twice.
@certik Let me know if this would be a correct approach